### PR TITLE
qemu.plugins: init at 6.2.0

### PR DESCRIPTION
**Pull request by erdnaxe**
###### Description of changes

When building QEMU, it is possible to build contributed plugins. These plugins can be loaded with QEMU `-plugin` parameter. More documentation about how to use these: https://qemu.readthedocs.io/en/latest/devel/tcg-plugins.html#example-plugins

I propose to add `make plugins` as a `postBuild` step (ninja does not have the target) and move the output to `qemu.plugins` output.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
On macos 10.15:

```
builder for '/nix/store/ajrkvkfwqw6z4nkww2m5c3yxcbx1jwr6-qemu-6.2.0.drv' failed with exit code 2; last 10 log lines:
  [6737/6738] Linking target tests/qtest/test-x86-cpuid-compat
  [6738/6738] Linking target tests/qtest/rtas-test
  /nix/store/8588rrhd0g7r3k7rf2x22m0l0ffg4xkk-ninja-1.10.2/bin/ninja  build.ninja && touch build.ninja.stamp
  ninja: no work to do.
  /nix/store/5sr2zsj8gm3a7jigjl0biqs62rsik5jv-meson-0.60.3/bin/meson introspect --targets --tests --benchmarks | /nix/store/3j4k5ja8zpr7v5xiyfgjwjb8qnl4iycv-python3-3.9.10/bin/python3 -B scripts/mtest2make.py > Makefile.mtest
    BUILD   example plugins
  ld: unknown option: -soname
  clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
  make[1]: *** [Makefile:39: libexeclog.so] Error 1
  make: *** [Makefile:173: plugins] Error 2
```

**Comment by erdnaxe** [comment]
> @risicle  On macos 10.15:
> 
> ```
> builder for '/nix/store/ajrkvkfwqw6z4nkww2m5c3yxcbx1jwr6-qemu-6.2.0.drv' failed with exit code 2; last 10 log lines:
>   [6737/6738] Linking target tests/qtest/test-x86-cpuid-compat
>   [6738/6738] Linking target tests/qtest/rtas-test
>   /nix/store/8588rrhd0g7r3k7rf2x22m0l0ffg4xkk-ninja-1.10.2/bin/ninja  build.ninja && touch build.ninja.stamp
>   ninja: no work to do.
>   /nix/store/5sr2zsj8gm3a7jigjl0biqs62rsik5jv-meson-0.60.3/bin/meson introspect --targets --tests --benchmarks | /nix/store/3j4k5ja8zpr7v5xiyfgjwjb8qnl4iycv-python3-3.9.10/bin/python3 -B scripts/mtest2make.py > Makefile.mtest
>     BUILD   example plugins
>   ld: unknown option: -soname
>   clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
>   make[1]: *** [Makefile:39: libexeclog.so] Error 1
>   make: *** [Makefile:173: plugins] Error 2
> ```

My bad, I should have realized that QEMU plugins Makefile does not support Darwin. It should now work as I added conditions on `stdenv.isLinux`.

File: pkgs/applications/virtualization/qemu/default.nix [hunk 1]
```diff
@@ -233,13 +235,21 @@ stdenv.mkDerivation rec {
...
+    done
   '' + lib.optionalString gtkSupport ''
     # wrap GTK Binaries
     for f in $out/bin/qemu-system-*; do
       wrapGApp $f
     done
   '';
   preBuild = "cd build";
+  # contrib/plugins/Makefile requires support for -soname option
+  postBuild = lib.optionals stdenv.isLinux "make plugins";
```
Comment (alyssais) [inline pkgs/applications/virtualization/qemu/default.nix:252]: Considering `make plugins` just runs `make` in the plugins directory, and the plugins directory says "This Makefile example is fairly independent from the main makefile", I wonder if it'd be better to skip the top-level Makefile and do `make -C plugins` here?  Also remember to pass the standard make flags variables. (From memory: `$makeFlags "${makeFlagsArray[@]}" $buildFlags "${buildFlagsArray[@]}"`, but do check.)

**Review by alyssais — COMMENTED** [review]
Can you share some information about your use case?  Throughout the source code, the plugins that come with QEMU are often described as "examples", which makes me a bit hesitant to include them with the package unless I know the rationale.

**Pull request closed without merging**